File: services/app/internal/domain/categories/category.go
package categories

type Category struct {
	Name           string
	BibFields      []field
	CiteFields     []field
	CitaviCategory string
	CitaviFilter   []string
}

type field struct {
	Name          string
	Format        format
	CitaviMapping []string
}

type format struct {
	Prefix       string
	Suffix       string
	Style        string
	Preformatted bool
}

const (
	categoriesFile = "/data/bib_categories.json"
)

File: services/app/frontend/src/domain/category/category.ts

export interface Category {
    Name: string
    CtiaviCategory: string
}

export interface field {
    Name: string
    Format: format
    CitaviMapping: string[]
}

export interface format {
    Prefix: string
    Suffix: string
    Style: string
    Preformatted: boolean
}
